olive/resource_path.py

```diff
@@ -335,15 +330,72 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
                 ml_client.models.download,
                 [self.config.name],
                 {"version": self.config.version, "download_path": temp_dir},
-                max_tries=self.config.azureml_client.max_operation_retries,
-                delay=self.config.azureml_client.operation_retry_interval,
+                max_tries=azureml_client_config.max_operation_retries,
+                delay=azureml_client_config.operation_retry_interval,
                 exceptions=ServiceResponseError,
             )
             new_path.parent.mkdir(parents=True, exist_ok=True)
             shutil.move(temp_dir / self.config.name / model_path.name, new_path)
         return str(new_path)
 
 
+class AzureMLModel(AzureMLResource):
+    """AzureML Model resource path."""
+
+    name = ResourceType.AzureMLModel
+
+    @staticmethod
+    def _default_config() -> Dict[str, Any]:
+        return {
+            "azureml_client": ConfigParam(
+                type_=AzureMLClientConfig, required=True, description="AzureML client config."
+            ),
+            "name": ConfigParam(type_=str, required=True, description="Name of the model."),
+            "version": ConfigParam(type_=Union[int, str], required=True, description="Version of the model."),
+        }
+
+    def get_path(self) -> str:
+        return f"azureml:{self.config.name}:{self.config.version}"
+
+    def get_aml_client_config(self) -> AzureMLClientConfig:
+        return self.config.azureml_client
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        ml_client = self.config.azureml_client.create_client()
+        return super().save_to_dir(dir_path, ml_client, self.config.azureml_client, overwrite, name)
+
+
+class AzureMLRegistryModel(AzureMLResource):
+    """AzureML Model resource path."""
+
+    name = ResourceType.AzureMLRegistryModel
+
+    @staticmethod
+    def _default_config() -> Dict[str, Any]:
+        return {
+            "azureml_client": ConfigParam(
+                type_=AzureMLClientConfig, required=False, description="AzureML client config."
+            ),
+            "registry_name": ConfigParam(type_=str, required=True, description="Name of the registry."),
+            "name": ConfigParam(type_=str, required=True, description="Name of the model."),
+            "version": ConfigParam(type_=Union[int, str], required=True, description="Version of the model."),
+        }
+
+    def get_path(self) -> str:
+        return (
+            f"azureml://registries/{self.config.registry_name}/models/{self.config.name}/versions/{self.config.version}"
+        )
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        # azureml client
+        azureml_client_config = self.config.azureml_client or AzureMLClientConfig()
+
+        ml_client = AzureMLClientConfig.create_registry_client(
```

Comment:
this doesn't really matter, but if we use `aml_client_config.create_registry_client`, we won't need to expose the `default_auth_params` parameter. 
This also makes the `create_client` and `create_registry_client` signatures similar and cleaner.

Reply:
I'm not sure if removing `default_auth_params` for registry client will work for some scenarios that needs it. I personally sometimes needs this arg for aml client config locally.

Reply:
oh, I didn't mean to remove it entirely. 
What I meant is since we already have an object `azureml_client_config`, it seems cleaner to just do 
```
ml_client = azureml_client_config.create_registry_client(self.config.registry_name)
```

This way, `create_registry_client` doesn't need the `default_auth_params` since it can get it from it's own `self.default_auth_params` like `create_client` does.

Reply:
Yeah I missed that we have a aml client config instance now. I'll update this.